### DISCUSSION TITLE
langiumLangiumDocuments.getOrCreateDocument return all properties except one properties

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-glsp/glsp#1157] [Q&A]
my requirement  ast root node is Ld like below second screenshot,when i call langiumLangiumDocuments.getOrCreateDocument(URI.parse(rootUri)) method,it return all properties(coils,ldnodes,locks) like below first screenshot except 'diagram',i want to know why(note:currently ,the corresponding file is an empty file)
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/a86f0f00-cebe-4a07-b7e8-79207b69f90d)
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/3c13f511-6991-4deb-bcb6-9176c5070897)
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/02a8277f-e46f-41a8-9b7a-d259cc6c81a9)

**Comment by planger** [comment]
I think, this is because the document is empty. If the document is empty, I assume, the array properties in `value` are initialized with empty arrays `[]`, but the object properties, such as `diagram` remain `undefined` until an actual value is set.

**Reply by zrwsmd** [reply]
i guess so,but how could it not be undefined?my requirement is when i drag a node(e.g:workflow tasknode)and save it ,corresponding file  have text content,but when i create a node like below screenshot,at moment this.state.diagramRoot is still undefined,how could i assign it
![image](https://github.com/eclipse-glsp/glsp/assets/26995040/258245aa-d9bb-49c1-8c39-3d8b052564c9)

**Reply by planger** [reply]
I'm afraid, this is hard for me to debug without stepping through the actual code.
